sechub-shared-kernel/src/main/java/com/daimler/sechub/sharedkernel/resilience/ResilientActionExecutor.java

```diff
@@ -13,198 +15,243 @@
+            throw e;
+        } else {
+            LOG.error("Returned proposal is not wellknown and so cannt be handled:{}", proposal.getClass().getName());
+            throw e;
+        }
+    }
+
+    private ResilienceProposal findFirstProposalFromConsultants(ResilienctActionContext context) {
+        ResilienceProposal proposal = null;
+        for (ResilienceConsultant consultant : consultants) {
+            proposal = consultant.consultFor(context);
+            if (proposal != null) {
+                break;
+            }
+        }
+        return proposal;
+    }
+
+    public void add(ResilienceConsultant consultant) {
+        Objects.requireNonNull(consultant, "consultant may not be null!");
+        consultants.add(consultant);
+    }
+
+    class ResilienctActionContext implements ResilienceContext {
+
+        private Exception currentError;
+        private int retriesCount;
+        private boolean retryNecessary;
+        private ResilienceCallback callBack;
+        private Map<String, Object> parameters = new HashMap<>(1);
+
+        public ResilienctActionContext(ResilienceCallback callBack) {
+            this.callBack = callBack;
+        }
+
+        public void prepareForNextCheck(Exception e) {
+            this.retryNecessary = false;
+            this.currentError = e;
+        }
+
+        public void forceFallThrough(FallthroughResilienceProposal proposal) {
+            this.retryNecessary = false;
+            fallThroughSupport.enable(currentError, proposal.getInfo(), proposal.getMillisecondsForFallThrough());
+        }
+
+        public boolean isRetryNecessary() {
+            return retryNecessary;
+        }
+
+        public void forceRetry() {
+            retryNecessary = true;
+            retriesCount++;
+        }
+
+        public void setCurrentError(Exception currentError) {
+            this.currentError = currentError;
+        }
+
+        @Override
+        public Exception getCurrentError() {
+            return currentError;
+        }
+
+        @Override
+        public int getAlreadyDoneRetries() {
+            return retriesCount;
+        }
+
+        /**
+         * @return callback or <code>null</code>
+         */
+        public ResilienceCallback getCallBack() {
+            return callBack;
+        }
+
+        @Override
+        public <V> void setValue(String key, V value) {
+            parameters.put(key, value);
+        }
+
+        @SuppressWarnings("unchecked")
+        public <V> V getValueOrNull(String key) {
+            Object v = parameters.get(key);
+            if (v == null) {
+                return null;
+            }
+            V result = null;
+            try {
+                result = (V) v;
+            } catch (ClassCastException e) {
+                LOG.error("Was not able to cast key:{} to wanted type. Was orignally: {}", key, v.getClass(), e);
+            }
+            return result;
+        };
```

Comment:
Is the semicolon at the end intentional?

Reply:
Will be fixed